### PR TITLE
Replace interface in custom normalizer

### PR DESCRIPTION
**Pull request by tony-tran**
Replace NormalizerInterface by ContextAwareNormalizerInterface to use `context` parameter in supportNormalization method

### CODE REVIEW

**Comment by tony-tran** [comment]
It's the same from 4.1 to 4.4

File: serializer/custom_normalizer.rst [hunk 1]
```diff
@@ -21,10 +21,10 @@ to customize the normalized data. To do that, leverage the ``ObjectNormalizer``:
 
     use App\Entity\Topic;
     use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
-    use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+    use Symfony\Component\Serializer\Normalizer\ContextAwareNormalizerInterface;
     use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
 
-    class TopicNormalizer implements NormalizerInterface
+    class TopicNormalizer implements ContextAwareNormalizerInterface
```
Comment (maxhelias) [inline serializer/custom_normalizer.rst]: If we make this change, we must also change the signature of the function `supportsNormalization` to accept the context in 3rd parameters

Comment (tony-tran) [reply serializer/custom_normalizer.rst]: it's exactly the reason why I propose to change the interface if we want to keep the context parameter in method.

in ContextAwareNormalizerInterface : 

```
public function supportsNormalization($data, $format = null, array $context = []);
```

In NormalizerInterface:

```    
public function supportsNormalization($data, $format = null);
```

If we want keep `NormalizerInterface` we need to remove the context in the example.

**Comment by javiereguiluz** [comment]
Thanks for this contribution Tony. Please note that if this change is approved, we'll merge it in 4.3 branch, the oldest maintained branch where this is applicable (see https://symfony.com/roadmap#maintained-symfony-branches).

**Comment by tony-tran** [comment]
My bad, this PR is for 4.2 minimum ....

**Comment by tony-tran** [comment]
Yes thank you @javiereguiluz

File: serializer/custom_normalizer.rst (hunk 1 above)
Comment (tony-tran) [reply serializer/custom_normalizer.rst]: Target change (4.1 to 4.3)

**Review by maxhelias — APPROVED** [review]

**Comment by OskarStark** [comment]
> Yes thank you @javiereguiluz. Do I have to change the target of the PR ?

No we take care of the switch, thank you 👍

**Comment by javiereguiluz** [comment]
Tony, thanks for fixing this and for rebasing the pull request to the right branch. Congrats also on your first Symfony Docs contribution!

**Pull request merged**